app/questionnaire/rules.py

```diff
@@ -156,6 +182,19 @@ def _get_answer_count_minus_one(filtered_answers):
     return min(max(len(filtered_answers) - 1, 0), MAX_REPEATS - 1)
 
 
+def _get_comparison_id_value(when_rule, answer_store, group_instance, group_instance_id):
+    """
+        Gets the value of an answer specified as an operand in a comparator ( i.e right hand argument ,
+        or rhs in if lhs>rhs ) In a when clause
+        Supports lhs being in a repeating group and rhs can be either in a similar sized repeating group or
+        not in a repeating group
+    """
+
+    if 'comparison_id' in when_rule:
+        return get_answer_store_value(answer_index=when_rule['comparison_id'], answer_store=answer_store,
+                                      group_instance=group_instance, group_instance_id=group_instance_id)
```

Comment:
I'm not sure this will always work if the comparison answer doesn't have the same group instance and group_instance_id as the group you're currently in. For instance if you're in a repeating group but the answer is outside of that repeating group then group_instance and group_instance_id may well not match. Alternatively if you're in a repeating group made up of more than one group drivers then the group_instance wouldn't be correct.
We need some logic to work out where the comparison_id sits in relation to the group you are currently in.

Reply:
Ah, that sounds right. I'll write some tests for that and fix it up.